apps/shadcn-example/src/components/wallet-connection/connect-wallet.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  AptosWallet,
+  WalletItem,
+  isInstallRequired,
```

Comment:
what is the end dev experience when using the shadcn selector modal? I would expect the user dont need to do everything we see here on the example dapp.

I can imagine something like, on the dapp side,
```
import {WalletSelector} from "@aptos-labs/wallet-adapter-react"
```
and then WalletSelector just holds everything (same as current antd), with probably properties we can provide to it.

Is this what you expect? or you have something else in mind?

Reply:
So right now, if you're using shadcn, you would pretty much just copy all the code from this directory into your app and ensure you install all the shadcn components it makes use of. Ideally, we would have an `@aptos-labs/wallet-adapter-shadcn` package that handles that for you, but since shadcn is a CLI and not an actual NPM package, I would have to think more about how that would work.

The goal of the `WalletItem` headless component and its associated utility functions is to make creating custom wallet selectors easier without asserting any design decisions. It's still a very low-level abstraction and there's a lot more we can build on top of it. The benefit is that you can reuse your own dialog implementation or choose not to use a dialog at all (for example, Find Out had a wallet connect page instead of a dialog).

I'll try to make this more clear in the documentation, but these headless components are building blocks for making a wallet adapter with your own design system rather than a single component you just drop in your app. In the future, we can use these building blocks to quickly create more packages like maybe an `@aptos-labs/wallet-adapter-ark-ui`, which would be closer to a single component that you drop into your app since Ark UI provides dialog, collapsible, and dropdown menu components.

Reply:
any reason not getting the example code as is into a@aptos-labs npm package, and have the end user simply import the WalletItem & co? there is a bigger vision to it and it sounds interesting but I think meanwhile we can wrap it all in a package to ease the end dev experience (without expecting the end dev to add more shadcn component to it, etc)

Reply:
I thought about this a bit over the weekend, and unfortunately I don't think it would make sense to wrap this shadcn wallet selector in a package. At the start of the [shadcn docs](https://ui.shadcn.com/docs#:~:text=This%20is%20NOT%20a%20component%20library.%20It%27s%20a%20collection%20of%20re%2Dusable%20components%20that%20you%20can%20copy%20and%20paste%20into%20your%20apps.), they state:

> This is NOT a component library. It's a collection of re-usable components that you can copy and paste into your apps.

Copy/Paste and code ownership are the core philosophies of shadcn. I was looking at 3rd party shadcn-based components and they all basically say "just copy this file to your components directory". Shadcn isn't like most UI tools, and serves more of a starting point for your UI rather than a be-all and end-all solution.

To make the process simpler, I put the `WalletSelector` component in a single ~150 line `wallet-selector.tsx` that users can copy into their app. Also, I updated the `README.md` of this app with step-by-step instructions. In those instructions I provide a single command they can run to ensure they have all the prerequisite components:

```bash
npx shadcn-ui@latest add button collapsible dialog dropdown-menu toast
```

If you think about these wallet adapter UIs as a sliding scale from simplicity to customizability, our MUI and AntD packages are at the far end of simplicity and the solution in this PR is at the far end of customizability. If you just want to drop a component into your app and call it a day, the MUI and AntD packages are still the go-to solutions. If you want to roll your own wallet selector with your app's styles and existing dialog component, then the `WalletItem` headless component and new utility functions will greatly simplify that.

With all that being said, once I finish all of the headless components for handling Aptos Connect, I do plan on adding a new package to this repo that sits at the middle of the scale between simplicity and customizability. It will export a single `WalletSelector` component and include it's own implementation of a dialog, dropdown menu, collapsible, etc. The goal is to eventually build up this mildly opinionated package to something like [Family ConnectKit](https://docs.family.co/connectkit/theming).

Reply:
This makes sense to me. Anyone already familiar with shadcn won't be daunted by these steps. In the absence of a CLI tool, perhaps in addition to linking to the file you could include a `curl` command a user could copy and paste. Just an idea.